reposerver/repository/repository.go

```diff
@@ -852,6 +776,85 @@ func (s *Service) runManifestGenAsync(ctx context.Context, repoRoot, commitSHA,
 	ch.responseCh <- manifestGenCacheEntry.ManifestResponse
 }
 
+func (s *Service) getRepoRefs(appSource *v1alpha1.ApplicationSource, srcRefs v1alpha1.RefTargetRevisionMapping, revision, commitSHA string) (map[string]repoRef, error) {
+	repoRefs := make(map[string]repoRef)
+
+	// Checkout every one of the referenced sources to the target revision before generating Manifests
+	for _, valueFile := range appSource.Helm.ValueFiles {
+		if strings.HasPrefix(valueFile, "$") {
+			refVar := strings.Split(valueFile, "/")[0]
+
+			refSourceMapping, ok := srcRefs[refVar]
+			if !ok {
+				if len(srcRefs) == 0 {
+					return map[string]repoRef{}, fmt.Errorf("source referenced %q, but no source has a 'ref' field defined", refVar)
```

Comment:
Most go code seems to `return nil, err` when the type is nullable.

```suggestion
					return nil, fmt.Errorf("source referenced %q, but no source has a 'ref' field defined", refVar)
```

Reply:
Didn't think the map was nullable, but it works so it's updated